ethstaker_deposit/cli/partial_deposit.py

```diff
@@ -0,0 +1,180 @@
+    validate_keystore_file,
+    validate_partial_deposit_amount,
+    validate_withdrawal_address,
+)
+
+
+FUNC_NAME = 'partial_deposit'
+
+
+@click.command(
+    help=load_text(['arg_partial_deposit', 'help'], func=FUNC_NAME),
+)
+@jit_option(
+    callback=captive_prompt_callback(
+        lambda x: closest_match(x, ALL_CHAIN_KEYS),
+        choice_prompt_func(
+            lambda: load_text(['arg_partial_deposit_chain', 'prompt'], func=FUNC_NAME),
+            ALL_CHAIN_KEYS
+        ),
+    ),
+    default=MAINNET,
+    help=lambda: load_text(['arg_partial_deposit_chain', 'help'], func=FUNC_NAME),
+    param_decls='--chain',
+    prompt=choice_prompt_func(
+        lambda: load_text(['arg_partial_deposit_chain', 'prompt'], func=FUNC_NAME),
+        ALL_CHAIN_KEYS
+    ),
+)
+@jit_option(
+    callback=captive_prompt_callback(
+        lambda file: validate_keystore_file(file),
+        lambda: load_text(['arg_partial_deposit_keystore', 'prompt'], func=FUNC_NAME),
+    ),
+    help=lambda: load_text(['arg_partial_deposit_keystore', 'help'], func=FUNC_NAME),
+    param_decls='--keystore',
+    prompt=lambda: load_text(['arg_partial_deposit_keystore', 'prompt'], func=FUNC_NAME),
+    type=click.Path(exists=True, file_okay=True, dir_okay=False),
+)
+@jit_option(
+    callback=captive_prompt_callback(
+        lambda x: x,
+        lambda: load_text(['arg_partial_deposit_keystore_password', 'prompt'], func=FUNC_NAME),
+        None,
+        lambda: load_text(['arg_partial_deposit_keystore_password', 'invalid'], func=FUNC_NAME),
+        True,
+    ),
+    help=lambda: load_text(['arg_partial_deposit_keystore_password', 'help'], func=FUNC_NAME),
+    hide_input=True,
+    param_decls='--keystore_password',
+    prompt=lambda: load_text(['arg_partial_deposit_keystore_password', 'prompt'], func=FUNC_NAME),
+)
+@jit_option(
+    callback=captive_prompt_callback(
+        lambda amount: validate_partial_deposit_amount(amount),
+        lambda: load_text(['arg_partial_deposit_amount', 'prompt'], func=FUNC_NAME),
+        default="32",
+    ),
+    default="32",
+    help=lambda: load_text(['arg_partial_deposit_amount', 'help'], func=FUNC_NAME),
+    param_decls='--amount',
+    prompt=lambda: load_text(['arg_partial_deposit_amount', 'prompt'], func=FUNC_NAME),
+)
+@jit_option(
+    callback=captive_prompt_callback(
+        lambda address: validate_withdrawal_address(None, None, address),
+        lambda: load_text(['arg_withdrawal_address', 'prompt'], func=FUNC_NAME),
+        lambda: load_text(['arg_withdrawal_address', 'confirm'], func=FUNC_NAME),
+        lambda: load_text(['arg_withdrawal_address', 'mismatch'], func=FUNC_NAME),
+        default="",
+    ),
+    default="",
+    help=lambda: load_text(['arg_withdrawal_address', 'help'], func=FUNC_NAME),
+    param_decls=['--withdrawal_address', '--execution_address', '--eth1_withdrawal_credentials'],
+    prompt=lambda: load_text(['arg_withdrawal_address', 'prompt'], func=FUNC_NAME),
```

Comment:
ack

Looks like I gotta do the same for the amount as well.

Reply:
Amount is unexpected. Why? This is usually only necessary when there is a confirm. In other cases it doesn't matter that the callback is called twice.

Reply:
I'm pretty confused by it as well and I wasn't able to follow the logic.

lets say 32, was passed it. It was properly converted to 32000000000
But then immediately 32000000000 was passed in and would fail.

I ran into something like this before but can't recall how I resolved it. This modification though does fix the issue completely.

Reply:
Fair enough! It's the conversion that causes it. Anything that has a validation logic that changes the amount will fail when it gets called twice. 32 to 32000000000 is fine, but then the second call to verify the final value fails. We have one other instance where this happens, there it's a conversion to bytes. For that one I added a note in the comment that the callback converts.